### PR TITLE
Add prompts to ask user about installing additional but required packages

### PR DESCRIPTION
**Pull request by dawiidio**
It's just a small change but can improve flow while using cli.

When using `devs devtools src/main.ts` with "native" OS transports (eg: usb or serial - not via dashboard and WebUSB) user must specify transport by passing one of the following options

```txt
-u, --usb             listen to Jacdac over USB (requires usb)
-s, --serial          listen to Jacdac over SERIAL (requires serialport)
-i, --spi             listen to Jacdac over SPI (requires rpio, experimental)
```

all the above options require additional packages, that's why now you can install required package without leaving current process. Cli will detect if any of native transports is enabled and if required package is not installed will prompt question like the below one:

```txt
...
Install package "serialport" (Y/n)?
Installing package "serialport" ...
Package "serialport" installed!
...
```

after installation normal process flow will be restored with installed dependencies ready to use.

### CODE REVIEW

**Comment by dawiidio** [comment]
@microsoft-github-policy-service agree

File: cli/src/packageInstaller.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    return new Promise((res) => {
+        resolve = res
+    });
+}
+
+export const spawnAsyncInstaller = (packageName: string) => {
+    let resolve: () => void
+    let reject: () => void
+
+    const installProcess = spawn('npm', ['install', '--save', '--no-workspaces', packageName], {
```
Comment (pelikhan) [inline cli/src/packageInstaller.ts:55]: I think we have some code to detect users that rely on yarn

File: cli/src/build.ts [hunk 2]
```diff
@@ -573,7 +591,7 @@ async function buildOnce(file: string, options: BuildOptions) {
             .forEach(fn => {
                 log(`  ${fn.name} (${prettySize(fn.size)})`)
                 fn.users.forEach(user =>
-                    debug(`    <-- ${resolver.posToString(user[0])}`)
+                    debug(`    <-- ${resolver.posToString(user[0])}`),
```
Comment (pelikhan) [inline cli/src/build.ts:594]: is this , added by prettier? it kind of add noise to the PR

File: cli/src/transport.ts [hunk 3]
```diff
@@ -37,22 +38,22 @@ function tryRequire(name: string) {
     return require(name)
 }
 
-function createSPI() {
-    log(`adding SPI transport (requires "rpio" package)`)
+async function createSPI() {
+    await askForPackageInstallation('rpio');
```
Comment (pelikhan) [inline cli/src/transport.ts:42]: We need a non-interactive mode in CI build mode.

**Comment by pelikhan** [comment]
This is a great start! My advice is to keep you PRs very focused on a single feature. If you find a bug along the way, move it out to a different PR so that it can be merged separately.

The formatting issue is annoying (the added ,) maybe we can fix this before this PR so that it is "just a formatting PR".

A few more comments:

- when the native package is missing and the cli is in vscode mode, we should send a side channel message to vscode so that the install flow is gone through vscode UI (future PR)
- need a flag to disable all interactions behavior for CI scenarios

👍 keep going!

File: cli/src/build.ts (hunk 2 above)
Comment (dawiidio) [reply cli/src/build.ts:594]: yeah, must be prettier. Do you want to remove it?

File: cli/src/packageInstaller.ts (hunk 1 above)
Comment (dawiidio) [reply cli/src/packageInstaller.ts]: sure, detection added

File: cli/src/transport.ts (hunk 3 above)
Comment (dawiidio) [reply cli/src/transport.ts:42]: Good point! Flag added globally to the cli in this commit https://github.com/microsoft/devicescript/pull/653/commits/ddf9c15d7f374d1a93745e3eabc5dfaac6702f54

**Comment by dawiidio** [comment]
@pelikhan 
> when the native package is missing and the cli is in vscode mode, we should send a side channel message to vscode so that the install flow is gone through vscode UI (future PR)

sure, I can add this communication in next pr. Maybe you know, when cli is in vscode mode it opens some additional socket channel to communicate with plugin?

File: cli/src/build.ts [hunk 4]
```diff
@@ -58,14 +58,14 @@ export function setupWebsocket() {
         require("websocket-polyfill")
         // @ts-ignore
         global.Blob = require("buffer").Blob
-        global.WebSocket.prototype.send = function (this: any, data: any) {
+        global.WebSocket.prototype.send = function(this: any, data: any) {
```
Comment (pelikhan) [inline cli/src/build.ts:61]: what kind of setting do you have in prettier? we should not have this kind of syntax formatting creeping in.

File: cli/src/build.ts (hunk 2 above)
Comment (pelikhan) [reply cli/src/build.ts:592]: Yes, in general this kind of formatting changes really add noise to a PR and makes the review harder. If you have user settings overriding the repo .prettierrc file, you should disable them.

**Comment by pelikhan** [comment]
> @pelikhan
> 
> > when the native package is missing and the cli is in vscode mode, we should send a side channel message to vscode so that the install flow is gone through vscode UI (future PR)
> 
> sure, I can add this communication in next pr. Maybe you know, when cli is in vscode mode it opens some additional socket channel to communicate with plugin?

Search for `export function sendEvent<T extends SideEvent>(` to see how to send an event from the CLI, and `subSideEvent<SideOutputEvent>` to receive that even on the vscode side

**Comment by pelikhan** [comment]
I've added a common linting task in package.json and also a dep on prettier so that we run the same version. You should be able to merge `origin/main` and it'll fix the whitespace/comma changes.

**Comment by pelikhan** [comment]
Merge `origin/main`, run `yarn install` and run `yarn lint` to run prettier

**Comment by pelikhan** [comment]
@dawiidio i've added a commit (https://github.com/microsoft/devicescript/commit/4372a1464ed26dc3e8c9f0e7a7bafc6b2987948c) to prep for this change. You can squeeze in the interactive piece of your changes in the TODO I left. Might be easier to start a new PR though.

**Comment by pelikhan** [comment]
Thank you @dawiidio , your PR has made it into https://github.com/microsoft/devicescript/pull/656

**Pull request closed without merging**